### PR TITLE
Capture exceptions in async ExecuteStatement

### PR DESCRIPTION
**Pull request by pan3793**
### _Why are the changes needed?_

When ExecuteStatement running in async, if engine broken(e.g. killed by yarn), the thread just crash without `setOperationException(ke)`, then client(like `beeline`) will hang.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/latest/tools/testing.html#running-tests) locally before make a pull request

Manual tested on our yarn cluster: kill yarn app when executing statement.
1. `beeline` connect hiveserver2. Result: exit with error
2. `beeline` connect kyuubi(before this patch). Result: hang
3. `beeline` connect kyuubi(after this patch). Result:  exit with error

### CODE REVIEW

File: kyuubi-main/src/main/scala/org/apache/kyuubi/operation/ExecuteStatement.scala [hunk 1]
```diff
@@ -91,8 +91,10 @@ class ExecuteStatement(
       isComplete = true
       remoteState match {
         case INITIALIZED_STATE | PENDING_STATE | RUNNING_STATE =>
-          isComplete = false
-          statusResp = client.GetOperationStatus(statusReq)
+          try {
```
Comment (yaooqinn) [inline kyuubi-main/src/main/scala/org/apache/kyuubi/operation/ExecuteStatement.scala:94]: shall we wrap the whole method？ there are still other places calling the `client`

**Comment by yaooqinn** [comment]
cc @ulysses-you, we shall fail 1.2.0-rc3 for this

**Comment by yaooqinn** [comment]
please link the target issue

**Comment by ulysses-you** [comment]
nice catch !

**Review by yaooqinn — APPROVED** [review]
LGTM
